### ISSUE TITLE
Need search feature

### ISSUE THREAD

**Issue by ahmerafzal1** [barteksc/AndroidPdfViewer#259]
So when conducting a search feature introduced? This is very important.

**Comment by barteksc** [comment]
I do not plan introducing it, but you may try and send pull request.

**Comment by 1stmetro** [comment]
@ahmerafzal1 i would look at opensource OCR and add this to the app your trying to build so then you can search via OCR since the pdf your app is viewing is a bitmap and it wont be simple to build your own search facility.

**Comment by ahmerafzal1** [comment]
@1stmetro yeah I know, when you get OCR please share with me. Thanks in advance

**Comment by 1stmetro** [comment]
I was tempted to play with this one, shouldn't be too difficult to add but its the zoomTo that might be harder to sort although its probably just a case of using the location their code returns and match the location on a zoomto and then draw a circle around the location rather than highlight as it would be easier.

https://github.com/tesseract-ocr/tesseract

**Comment by barteksc** [comment]
Using OCR for such task is madness. Pdfium has methods for getting text from pages

**Comment by 1stmetro** [comment]
Ah sorry been to busy to investigate pdfium but now you have said that it will point people towards this issue and hopefully aim towards sorting a solution.

**Comment by odmfl** [comment]
pdfium should have this function, the problem is to implement it for android.  I remember the old app "google pdf viewer" which however was based on pdfium and had this function

I did some research and found some libraries that have implemented the search, but I don't know if they work, they should be tested.
It would take someone to test them and maybe fork and keep an updated fork.

Test them and let me know if they work in any way

https://github.com/gulshan98185/AndroidPdfViewer

https://github.com/beeline09/AdvancedPdfView

https://github.com/beeline09/android-support-pdfium

https://github.com/benjinus/android-support-pdfium
